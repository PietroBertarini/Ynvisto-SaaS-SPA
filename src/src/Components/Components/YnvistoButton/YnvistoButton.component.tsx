import React from 'react';
import { Button } from '@material-ui/core';
import { useStyles } from './YnvistoButton.styles';
import { SpinnerContainer } from '../Spinner/spinner.styles';

export enum EButtonColor {
  PRIMARY = 'PRIMARY',
  DANGER = 'DANGER',
}

type ButtonProps = {
  title: string,
  isLoading?: boolean,
  status?: string,
  underButtonText?: string,
  onClick: () => any
}

const YnvistoButton = ({
  title, onClick, isLoading, status, underButtonText,
} : ButtonProps) => {
  const classes = useStyles(status ? { status } : { status: 'PRIMARY' });
  const spinnerColor = (status === 'DANGER') ? '#EB5050' : '#E9C46A';

  return (
    <div className={classes.main}>
      <Button
        className={classes.button}
        variant="outlined"
        onClick={onClick}
      >
        {isLoading ? <SpinnerContainer size={30} spinnerColor={spinnerColor} /> : title }
      </Button>
      <h5 className={classes.underText}>{underButtonText}</h5>
    </div>
  );
};

export default YnvistoButton;
