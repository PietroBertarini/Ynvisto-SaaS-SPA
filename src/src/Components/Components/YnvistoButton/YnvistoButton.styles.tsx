import { createStyles, makeStyles } from '@material-ui/core/styles';

const styledBy = (property, mapping) => (props) => mapping[props[property]];
export const useStyles = makeStyles(() => createStyles({
  button: {
    color: styledBy('status', {
      PRIMARY: '#E9C46A',
      DANGER: '#EB5050',
    }),
    borderColor: styledBy('status', {
      PRIMARY: '#E9C46A',
      DANGER: '#EB5050',
    }),
    fontSize: 20,
    textTransform: 'none',
    background: 'transparent',
    borderRadius: 20,
    float: 'right',
    marginRight: '3vw',
    padding: '5px 20px',
    borderWidth: 3,
    marginTop: '2vw',
    height: '3vw',
    minWidth: '12vw',
    '&:hover': {
      backgroundColor: 'rgba(233, 196, 106, 0.14)',
      color: styledBy('status', {
        CONFIRM: '#E9C46A',
        REMOVE: '#EB5050',
      }),
      borderColor: styledBy('status', {
        CONFIRM: '#E9C46A',
        REMOVE: '#EB5050',
      }),
    },
  },
  underText: {
    fontSize: 18,
    color: '#fff',
    fontWeight: 'bold',
    marginRight: '3vw',
  },
  main: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
}));
