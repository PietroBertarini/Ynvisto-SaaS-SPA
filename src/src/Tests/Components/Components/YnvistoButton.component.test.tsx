import React from 'react';
import { shallow } from 'enzyme';
import YnvistoButton from '../../../Components/Components/YnvistoButton/YnvistoButton.component';

test('Renders YnvistoButton without status prop component and match to previously snapshot', () => {
  // Empty function, according by ESlint documentation, to avoid empty arrow functions
  function emptyFunc() {
    // do nothing
  }

  const component = shallow(
    <YnvistoButton title="" isLoading={false} onClick={emptyFunc} />,
  );
  expect(component).toMatchSnapshot();
});

test('Renders YnvistoButton with status prop component and match to previously snapshot', () => {
  // Empty function, according by ESlint documentation, to avoid empty arrow functions
  function emptyFunc() {
    // do nothing
  }

  const component = shallow(
    <YnvistoButton title="" status="DANGER" isLoading={false} onClick={emptyFunc} />,
  );
  expect(component).toMatchSnapshot();
});
