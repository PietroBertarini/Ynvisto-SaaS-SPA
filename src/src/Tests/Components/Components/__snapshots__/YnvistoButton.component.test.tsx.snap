// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders YnvistoButton with status prop component and match to previously snapshot 1`] = `
<div
  className="makeStyles-main-3"
>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-1 makeStyles-button-5"
    onClick={[Function]}
    variant="outlined"
  />
  <h5
    className="makeStyles-underText-2"
  />
</div>
`;

exports[`Renders YnvistoButton without status prop component and match to previously snapshot 1`] = `
<div
  className="makeStyles-main-3"
>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-1 makeStyles-button-4"
    onClick={[Function]}
    variant="outlined"
  />
  <h5
    className="makeStyles-underText-2"
  />
</div>
`;
